### PR TITLE
Feat/data stuff

### PR DESCRIPTION
**Pull request by thestonefox**

### CODE REVIEW

**Comment by thestonefox** [comment]
These two components are the prerequisite to allowing the mouse scroll wheel to control the rotation of objects (e.g. like the pointer direction indicator).

**Comment by thestonefox** [comment]
Solution goes:

```
UnityAxis1DAction: AxisName[Mouse ScrollWheel] -> ValueChanged ---> AngleToVector2Direction.Transform

AngleToVector2Direction.Transformed ---> Vector2ComponentExtractor.Extract [on X]
AngleToVector2Direction.Transformed ---> Vector2ComponentExtractor.Extract [on Y]

Vector2ComponentExtractor.Extracted [X] ---> FloatAction.Receive (X Axis)
Vector2ComponentExtractor.Extracted [Y] ---> FloatAction.Receive (Y Axis)
```

These new X Axis and Y Axis actions can be fed into the `AxisRotator` to rotate an object with the mouse scroll wheel.

File: Runtime/Data/Operation/Extraction/Vector2ComponentExtractor.cs [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        public Vector2 Source { get; set; }
+        /// <summary>
+        /// The component to extract from the <see cref="Vector2"/>.
+        /// </summary>
+        [Serialized]
+        [field: DocumentedByXml]
+        public Vector2Component ComponentToExtract { get; set; } = Vector2Component.X;
+
+        /// <summary>
+        /// Emitted when the <see cref="float"/> component from <see cref="Source"/> extracted.
```
Comment (bddckr) [inline Runtime/Data/Operation/Extraction/Vector2ComponentExtractor.cs]:
```suggestion
        /// Emitted when the <see cref="float"/> component from <see cref="Source"/> is extracted.
```

File: Runtime/Data/Operation/Extraction/Vector2ComponentExtractor.cs [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        /// <returns>The extracted <see cref="float"/>.</returns>
+        public virtual float? Extract()
+        {
+            if (!isActiveAndEnabled)
+            {
+                Result = null;
+                return null;
+            }
+
+            Result = null;
```
Comment (bddckr) [inline Runtime/Data/Operation/Extraction/Vector2ComponentExtractor.cs]:
```suggestion
```

File: Runtime/Data/Operation/Extraction/Vector2ComponentExtractor.cs [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+            {
+                case Vector2Component.X:
+                    Result = Source.x;
+                    break;
+                case Vector2Component.Y:
+                    Result = Source.y;
+                    break;
+            }
+
+            Extracted?.Invoke((float)Result);
```
Comment (bddckr) [inline Runtime/Data/Operation/Extraction/Vector2ComponentExtractor.cs]:
```suggestion
            Extracted?.Invoke(Result.Value);
```

File: Runtime/Data/Type/Transformation/Conversion/AngleToVector2Direction.cs [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        /// <param name="input">The value to transform.</param>
+        /// <returns>The transformed value.</returns>
+        protected override Vector2 Process(float input)
+        {
+            angle += input;
+            angle = Mathf.Repeat(angle, 360f);
+            outputAngle.x = Mathf.Cos(angle * Mathf.Deg2Rad);
+            outputAngle.y = Mathf.Sin(angle * Mathf.Deg2Rad);
+            Direction = outputAngle;
+            Direction *= 1 / Mathf.Max(Mathf.Abs(Direction.x), Mathf.Abs(Direction.y));
```
Comment (bddckr) [inline Runtime/Data/Type/Transformation/Conversion/AngleToVector2Direction.cs]:
```suggestion
            Direction *= 1f / Mathf.Max(Mathf.Abs(Direction.x), Mathf.Abs(Direction.y));
```

**Pull request merged**